File: app/src/main/java/com/nvvi9/giphyme/domain/mappers/GifItemMapper.java
package com.nvvi9.giphyme.domain.mappers;

import com.nvvi9.giphyme.data.BaseMapper;
import com.nvvi9.giphyme.data.preview.GifPreview;
import com.nvvi9.giphyme.vo.GifItem;

public class GifItemMapper implements BaseMapper<GifPreview, GifItem> {

    @Override
    public GifItem map(GifPreview value) {
        return new GifItem(value.getId(), value.getTitle(), value.getUrl(), value.getPreviewUrl(), value.getInFavorites());
    }
}

File: app/src/main/java/com/nvvi9/giphyme/domain/mappers/GifItemMapper.kt
package com.nvvi9.giphyme.domain.mappers

import com.nvvi9.giphyme.data.BaseMapper
import com.nvvi9.giphyme.data.preview.GifPreview
import com.nvvi9.giphyme.vo.GifItem

class GifItemMapper : BaseMapper<GifPreview, GifItem> {

    override fun map(value: GifPreview): GifItem =
        GifItem(value.id, value.title, value.url, value.previewUrl, value.inFavorites)
}